Application/src/view/visuals/Draw.java

```diff
@@ -291,13 +290,45 @@ private void renderClass() {
      * Places the classes in the class diagram
      */
 
-    void renderClassDiagramClass(){
+    void renderClassDiagramClass() {
         if (allClassDiagramClasses.size() == 0) return;
-        int space = ((int)this.canvas_class.getWidth())/this.allClassDiagramClasses.size();
-        for(int i = 0; i < allClassDiagramClasses.size(); i++) {
-            int x = (space/2) + (i * space);
-            int y = 50 + (space/2)/4;
-            allClassDiagramClasses.get(i).place(new Coordinates(x,y), (space/2));
+        int space = ((int) this.canvas_class.getWidth()) / this.allClassDiagramClasses.size();
+        //The size of the matrix structure, i.e the amount of elements/classes
+        int matrixSize = allClassDiagramClasses.size();
+        int col = 3; //The "up to" -number of columns in the matrix, counting from 0.
+        //Rows depending on size of matrix
+        int rows;
+        //Find how many rows in this matrix
+        //If the size is less than the allowed column number, there is only one row
+        if(matrixSize <= col){ rows = 1;}
+        //else if the size is dividable by 3, then that's how many rows there should be
+        else  if (matrixSize % col == 0) { rows = matrixSize / col; }
+        //else, there should be +1 row, but not a full one
+        else { rows = ((matrixSize / col) + 1); }
+
```

Comment:
use the same Y for both of the cases. e.g.
int y = (int) ((this.canvas_class.getHeight() / (allClassDiagramClasses.size()/(col))));

Then just in the for loop for rows you have y*r.